src/sentry/models/event.py

```diff
@@ -320,47 +310,205 @@ def site(self):
+
+class SnubaEvent(EventCommon):
+    """
+        An event backed by data stored in snuba.
+
+        This is a readonly event and does not support event creation or save.
+        The basic event data is fetched from snuba, and the event body is
+        fetched from nodestore and bound to the data property in the same way
+        as a regular Event.
+    """
+
+    # The list of columns that we should request from snuba to be able to fill
+    # out the object.
+    selected_columns = [
+        'event_id',
+        'project_id',
+        'message',
+        'title',
+        'location',
+        'culprit',
+        'timestamp',
+        'group_id',
+        'platform',
+
+        # Required to provide snuba-only tags
+        'tags.key',
+        'tags.value',
+
+        # Required to provide snuba-only 'user' interface
+        'user_id',
+        'username',
+        'ip_address',
+        'email',
+    ]
+
+    __repr__ = sane_repr('project_id', 'group_id')
+
+    @classmethod
+    def get_event(cls, project_id, event_id):
+        from sentry.utils import snuba
+        result = snuba.raw_query(
+            start=datetime.utcfromtimestamp(0),  # will be clamped to project retention
+            end=datetime.utcnow(),  # will be clamped to project retention
+            selected_columns=cls.selected_columns,
+            filter_keys={
+                'event_id': [event_id],
+                'project_id': [project_id],
+            },
+        )
+        if 'error' not in result and len(result['data']) == 1:
+            return SnubaEvent(result['data'][0])
+        return None
+
+    def __init__(self, snuba_values):
+        assert set(snuba_values.keys()) == set(self.selected_columns)
+
+        self.__dict__ = snuba_values
+
+        # This should be lazy loaded and will only be accessed if we access any
+        # properties on self.data
+        node_id = SnubaEvent.generate_node_id(self.project_id, self.event_id)
+        self.data = NodeData(None, node_id, data=None)
+
+    # ============================================
+    # Snuba-only implementations of properties that
+    # would otherwise require nodestore data.
+    # ============================================
+    @property
+    def tags(self):
+        """
+        Override of tags property that uses tags from snuba rather than
+        the nodestore event body. This might be useful for implementing
+        tag deletions without having to rewrite nodestore blobs.
+        """
+        keys = getattr(self, 'tags.key', None)
+        values = getattr(self, 'tags.value', None)
+        if keys and values and len(keys) == len(values):
+            return sorted(zip(keys, values))
+        return []
+
+    def get_interface(self, name):
+        """
+        Override of interface getter that lets us return some interfaces
+        directly from Snuba data.
+        """
+        if name in ['user']:
+            from sentry.interfaces.user import User
```

Comment:
should return `None` if the columns are `None` for consistency with the nodestore path.